src/textual/widgets/_option_list.py

```diff
@@ -789,6 +842,22 @@ def get_option(self, option_id: str) -> Option:
                 f"There is no option with an ID of '{option_id}'"
             ) from None
 
+    def get_option_index(self, option_id):
```

Comment:
If we're introducing this as part of the public interface, and in service of `replace_option_prompt`, I think I would reduce the duplication of code by tweaking `remove_option` to be based on it too. I think it makes sense to deduplicate the code in any developer-facing method that's unlikely to be called frequently.

Reply:
I should mention, same for other similar developer-facing methods. If that's what #2986 was in reference to them I'm in favour of that and would support it.